langfuse/src/langfuse.ts

```diff
@@ -22,9 +25,33 @@ export class Langfuse extends LangfuseCore {
   private _storageCache: any;
   private _storageKey: string;
 
-  constructor(params: { publicKey: string; secretKey: string } & LangfuseOptions) {
+  constructor(params?: { publicKey: string; secretKey: string } & LangfuseOptions) {
+    // if params are not provided and environment variables exist, then retrieve data from the environment variables. 
+    if (!params) {
+      if (process.env.LANGFUSE_PUBLIC_KEY && process.env.LANGFUSE_SECRET_KEY) {
+        params = { publicKey: process.env.LANGFUSE_PUBLIC_KEY, secretKey: process.env.LANGFUSE_SECRET_KEY }
+      }
+      else {
+        throw new Error("You must pass your Langfuse project's api public key and secret key either within the constructor or as environment variables ");
+      }
+    }
+
+    if (!params.publicKey && process.env.LANGFUSE_PUBLIC_KEY) {
+      params.publicKey = process.env.LANGFUSE_PUBLIC_KEY
+    }
+
+    if (!params.secretKey && process.env.LANGFUSE_SECRET_KEY) {
+      params.secretKey = process.env.LANGFUSE_SECRET_KEY
+    }
+
+    if (!params.baseUrl && process.env.LANGFUSE_HOST) {
+      params.baseUrl = process.env.LANGFUSE_HOST
+    }
+
```

Comment:
i am confuse a bit , from
 ```ts
 if (!params.publicKey && process.env.LANGFUSE_PUBLIC_KEY) {
      params.publicKey = process.env.LANGFUSE_PUBLIC_KEY
    }

  if (!params.secretKey && process.env.LANGFUSE_SECRET_KEY) {
      params.secretKey = process.env.LANGFUSE_SECRET_KEY
    }

  if (!params.baseUrl && process.env.LANGFUSE_HOST) {
      params.baseUrl = process.env.LANGFUSE_HOST
    }
//☝️  if the user provides credentials in both the environment variables and the constructor, the code gives priority to the constructor values, 
```
to 

```ts

    params.publicKey =  process.env.LANGFUSE_PUBLIC_KEY || params.publicKey
    params.secretKey =  process.env.LANGFUSE_SECRET_KEY || params.secretKey
    params.baseUrl =  process.env.LANGFUSE_HOST || params.baseUrl;

//☝️ This code is now more readable. The key difference is that if the user provides credentials in both the environment variables and the constructor, the code gives priority to the environment variables, 
```

please read the comment 

 i can do the same in langfuseWeb

Reply:
Hi, thanks for the comment. We should give priority to the constructor variables.

Reply:
In this case, do we even need L39-L49?

Reply:
I think yes. Beacuse If we do not check the PK and SK in the environment variable, how can we use them? What you think ?